### PR TITLE
feat : add devise integration

### PR DESCRIPTION
**Pull request by nilp06**
1) generated devise model for User
2) added extra attributes for user model username , role  and mobile-number.
3) generated devise views.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
```
Comment (dhruvika29kevat) [inline app/models/user.rb:4]: Remove unnecessary comments.

File: app/models/user.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+         :recoverable, :rememberable, :validatable
+  enum :role, %i[freelancer client]
+  validates :role, presence: true
+  validates :user_name, presence: true
```
Comment (dhruvika29kevat) [inline app/models/user.rb:8]: Move this presence validation in one line.

File: app/models/user.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
+  enum :role, %i[freelancer client]
```
Comment (dhruvika29kevat) [inline app/models/user.rb:6]: remove the unnecessary colon before 'role' review the syntax

**Review by dhruvika29kevat — COMMENTED** [review]
Keep focus on optimisation

File: app/models/user.rb (hunk 1 above)
Comment (nilp06) [reply app/models/user.rb:2]: ok

File: app/models/user.rb (hunk 2 above)
Comment (nilp06) [reply app/models/user.rb]: ok

File: app/models/user.rb (hunk 3 above)
Comment (nilp06) [reply app/models/user.rb:4]: ok

**Pull request merged**